### DISCUSSION TITLE
Requirements for the Jupyter support in the Progress "roadmap"

### DISCUSSION THREAD

**Discussion by hyiltiz** [nushell/nushell#9299] [General]
Currently, the Progress section of Readme.md states Jupyter support is in prototype state. Searching `jupyter` in this repo returns only that line in `Readme.md`. 

What is expected for a full Jupyter support from Nu 1.0? I doubt hardly anyone expects Nu to render Jupyter notebooks or even simulate Jupyter engines itself (as we do not need a Jupyter replacement). Do we expect to have full support for:
- [ ] starting Jupyter (Lab) server
- [ ] seamless creation, loading and unloading virtual environments for backends supported by Jupyter, including Python, R, ...
- [ ] sending the server to background, and reviving it to foreground

I think the first item is supported already. https://github.com/nushell/nu_scripts/pull/513 helps item 2 for Python. Item 3 is probably out of scope for Jupyter.

**Comment by fdncred** [comment]
This is long out of date but was what we used to use for jupyter https://github.com/nushell/nu_jupyter

**Comment by hyiltiz** [comment]
@fdncred Thank you for pointing to nu_jupyter. I see that none of items in my check list covered what was intended by "Jupyter support" -- it meant having a `nu` kernel that integrates well with Jupyter.

What would be the intended use case for such a kernel?

**Reply by fdncred** [reply]
> What would be the intended use case for such a kernel?

The intent of this prototype was to run nushell within the jupyter ecosystem and have some of the functionality that jupyter provides. 

The prototype was a success but there was very little interest from the community in improving it, which is why it was never upgraded. We'd love for someone to take charge and reinvigorate this jupyter plugin experience.

**Reply by hyiltiz** [reply]
To help bring some more attention and gather more interest, it may be worthwhile to list features that may be useful for a (typical *and* a-typical) `nushell` user. A few I can think of:

- UI widgets that can be manipulated for visualization or IO actions
- Jupyter/Jupyterlab extensions (e.g. snippet support)
- Literate programming support for `nushell`, suitable for story telling and/or explorative work
- Plots and visualization (this may require using external tools for the visualization)

**Reply by fdncred** [reply]
I wonder if this could help https://github.com/oovm/jupyter-protocol
